commit 252d5ab4ffb4a6a2a20bf9a23bc392617849cc8e
Author: LorD ClockaN <davor@losinj.com>
Date:   Wed Jun 15 21:08:28 2011 +0200

    ADD: BFS: Allow non priviledged tasks to bounce between SCHED_BATCH/NORMAL policies
    
        This is necessary for Android so that the VM can bump priorities of batch
        threads around GC to avoid an inversion on GC

diff --git a/kernel/sched_bfs.c b/kernel/sched_bfs.c
index 7a69d50..b6469a9 100644
--- a/kernel/sched_bfs.c
+++ b/kernel/sched_bfs.c
@@ -3869,6 +3869,11 @@ recheck:
 				case SCHED_BATCH:
 					if (policy == SCHED_BATCH)
 						goto out;
+					/*
+					*ANDROID: Allow tasks to move between
+					*/
+					if (policy == SCHED_NORMAL)
+					break;
 					if (policy != SCHED_IDLEPRIO)
 						return -EPERM;
 					break;
